src/workspaces/features/WorkspaceConnectionPicker/WorkspaceConnectionPicker.tsx

```diff
@@ -85,7 +85,7 @@ const WorkspaceConnectionPicker = (props: WorkspaceConnectionPickerProps) => {
       loading={loading}
       withPortal={withPortal}
       displayValue={displayValue}
-      by="name"
+      by="id"
```

Comment:
since you use the slug just above, it's better to use it as well here.
```suggestion
      by="slug"
```